src/main/java/com/google/sps/servlets/GenerateServlet.java

```diff
@@ -28,14 +30,162 @@
 
   private static final String FILTER_ARRAY = "clicked-array";
 
+  String[] allPlaces = {"Paris", "New York City", "San Francisco", "London", "Sydney", "Venice"};
+  String[] allDifficulties = {"Easy", "Medium", "Hard"};
+
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    HashSet<String> userPlaces = new HashSet();
+    HashSet<String> userDifficulties = new HashSet();
+
+    ArrayList<Destination> allDestinations = new ArrayList();
+    // ArrayList<Name of Destination>
+    ArrayList<String> filteredDestinations = new ArrayList();
+
     // Get array of clicked filters and convert to ArrayList<String>
-    Gson gson = new Gson();
     HashSet<String> clickedFilters =
-        gson.fromJson(request.getParameter(FILTER_ARRAY), HashSet.class);
+        new Gson().fromJson(request.getParameter(FILTER_ARRAY), HashSet.class);
+
+    allDestinations = createFakeDestinations();
+    userPlaces = separateFilters(clickedFilters, allPlaces);
+    userDifficulties = separateFilters(clickedFilters, allDifficulties);
+    filteredDestinations = filterPlaces(userPlaces, allDestinations);
```

Comment:
For readability, I would suggest putting the variable declarations in-line with where they are assigned.  There isn't any value in declaring your variables as empty arrays and then assigning them values 9 lines down.   It makes it hard to follow the logic. 

```suggestion
    ArrayList<String> filteredDestinations = filterPlaces(userPlaces, allDestinations);
```